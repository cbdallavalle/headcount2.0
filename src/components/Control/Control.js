import React from 'react';
import PropTypes from 'prop-types';
import './Control.css'

const Control = (props) => {
  return (
    <div className="Control">
      <div className="input-cont">
        <label htmlFor="search"></label>
        <i className="fa fa-search search-icon" aria-hidden="true"></i>
        <input type="text"
               id="search"
               placeholder="Search by School"
               onChange={(e) => props.handleSearch(e)}
        />
      </div>
      <h1>Education Colorado</h1>
    </div>
  )
}

Control.propTypes = {
  handleSearch: PropTypes.func.isRequired
}

export default Control;