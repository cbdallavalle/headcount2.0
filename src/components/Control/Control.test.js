import React from 'react';
import { shallow } from 'enzyme';
import Control from './Control';

it('should match the snapshot', () => {
  const renderedComponent = shallow(<Control
    handleSearch={()=>{}} 
    searchError={false}
    resetDistrict={()=>{}}
  />);
  expect(renderedComponent).toMatchSnapshot();
});