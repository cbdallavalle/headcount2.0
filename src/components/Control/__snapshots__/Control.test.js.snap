// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Control
    handleSearch={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="input-cont"
>
          <label
                    htmlFor="search"
          />
          <i
                    aria-hidden="true"
                    className="fa fa-search"
          />
          <input
                    id="search"
                    onChange={[Function]}
                    placeholder="Search by School"
                    type="text"
          />
          <label
                    htmlFor="submit"
          />
          <input
                    id="submit"
                    type="submit"
          />
</div>,
        <h1>
          Education Colorado
</h1>,
      ],
      "className": "Control",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label
              htmlFor="search"
/>,
            <i
              aria-hidden="true"
              className="fa fa-search"
/>,
            <input
              id="search"
              onChange={[Function]}
              placeholder="Search by School"
              type="text"
/>,
            <label
              htmlFor="submit"
/>,
            <input
              id="submit"
              type="submit"
/>,
          ],
          "className": "input-cont",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "htmlFor": "search",
            },
            "ref": null,
            "rendered": null,
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "aria-hidden": "true",
              "className": "fa fa-search",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "search",
              "onChange": [Function],
              "placeholder": "Search by School",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "htmlFor": "submit",
            },
            "ref": null,
            "rendered": null,
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "submit",
              "type": "submit",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Education Colorado",
        },
        "ref": null,
        "rendered": "Education Colorado",
        "type": "h1",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="input-cont"
>
            <label
                        htmlFor="search"
            />
            <i
                        aria-hidden="true"
                        className="fa fa-search"
            />
            <input
                        id="search"
                        onChange={[Function]}
                        placeholder="Search by School"
                        type="text"
            />
            <label
                        htmlFor="submit"
            />
            <input
                        id="submit"
                        type="submit"
            />
</div>,
          <h1>
            Education Colorado
</h1>,
        ],
        "className": "Control",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                htmlFor="search"
/>,
              <i
                aria-hidden="true"
                className="fa fa-search"
/>,
              <input
                id="search"
                onChange={[Function]}
                placeholder="Search by School"
                type="text"
/>,
              <label
                htmlFor="submit"
/>,
              <input
                id="submit"
                type="submit"
/>,
            ],
            "className": "input-cont",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "htmlFor": "search",
              },
              "ref": null,
              "rendered": null,
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "aria-hidden": "true",
                "className": "fa fa-search",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "search",
                "onChange": [Function],
                "placeholder": "Search by School",
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "htmlFor": "submit",
              },
              "ref": null,
              "rendered": null,
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "submit",
                "type": "submit",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Education Colorado",
          },
          "ref": null,
          "rendered": "Education Colorado",
          "type": "h1",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
